gcore/gdal_priv.h

```diff
@@ -4321,13 +4321,41 @@ GDALDataset *GDALCreateOverviewDataset(GDALDataset *poDS, int nOvrLevel,
 // Should cover particular cases of #3573, #4183, #4506, #6578
 // Behavior is undefined if fVal1 or fVal2 are NaN (should be tested before
 // calling this function)
-template <class T> inline bool ARE_REAL_EQUAL(T fVal1, T fVal2, int ulp = 2)
+
+// The expression `abs(fVal1 + fVal2)` looks strange; is this a bug?
+// Should this be `abs(fVal1) + abs(fVal2)` instead?
+inline bool ARE_REAL_EQUAL(float fVal1, float fVal2, int ulp = 2)
 {
+    using std::abs;
     return fVal1 == fVal2 || /* Should cover infinity */
-           std::abs(fVal1 - fVal2) < std::numeric_limits<float>::epsilon() *
-                                         std::abs(fVal1 + fVal2) * ulp;
+           abs(fVal1 - fVal2) <
+               std::numeric_limits<float>::epsilon() * abs(fVal1 + fVal2) * ulp;
+}
+
+// We are using `std::numeric_limits<float>::epsilon()` for backward
+// compatibility
+inline bool ARE_REAL_EQUAL(double dfVal1, double dfVal2, int ulp = 2)
+{
+    using std::abs;
+    return dfVal1 == dfVal2 || /* Should cover infinity */
+           abs(dfVal1 - dfVal2) < std::numeric_limits<float>::epsilon() *
+                                      abs(dfVal1 + dfVal2) * ulp;
 }
 
+// The generic function above does not work for _Float16 because
+// std::numeric_limits<float>::epsilon() is the wrong choice
+#ifndef SWIGPYTHON
```

Comment:
Why testing for SWIGPYTHON outside of swig/include? That doesn't seem appropriate

Reply:
SWIG examines this file and is getting confused because it does not know `_Float16`. Is there a different way to tell SWIG to ignore this function? (I'm not a SWIG expert.)

Reply:
The error is
```
In file included from extensions/gdal_array_wrap.cpp:3395:
/localhome/eschnett/src/gdal/gcore/gdal_priv.h:4414:28: error: ‘bool ARE_REAL_EQUAL’ redeclared as different kind of entity
 4414 | inline bool ARE_REAL_EQUAL(_Float16 hfVal1, _Float16 hfVal2, int ulp = 2)
      |                            ^~~~~~~~
```